### PR TITLE
Create RecordDeclarationOrganizer

### PR DESCRIPTION
**Pull request by Youssef1313**

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/Organizing/OrganizeTypeDeclarationTests.cs [hunk 1]
```diff
@@ -20,276 +20,300 @@ namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Organizing
...
+}}";
 
             var final =
-@"class C {
+$@"{typeKind} C {{
     int A;
     int B;
     int C;
-}";
+}}";
```
Comment (Youssef1313) [inline src/EditorFeatures/CSharpTest/Organizing/OrganizeTypeDeclarationTests.cs:60]: @CyrusNajmabadi This looks like a nice feature that I was never aware it's existed. Can I ask where this feature is in Visual Studio? I can't find any option along the words "organize declarations"

**Comment by CyrusNajmabadi** [comment]
@Youssef1313 tests are failing :)

**Comment by Youssef1313** [comment]
@CyrusNajmabadi Fixed the failing test. Is this ready to merge?

File: src/Features/CSharp/Portable/Organizing/Organizers/RecordDeclarationOrganizer.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                syntax.Identifier,
+                syntax.TypeParameterList,
+                syntax.ParameterList,
+                syntax.BaseList,
+                syntax.ConstraintClauses,
+                syntax.OpenBraceToken,
+                MemberDeclarationsOrganizer.Organize(syntax.Members, cancellationToken),
+                syntax.CloseBraceToken,
+                syntax.SemicolonToken);
+        }
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Organizing/Organizers/RecordDeclarationOrganizer.cs:40]: can you jsut do `syntax.WithModifiers(...).WithMembers(...);`?

Comment (Youssef1313) [reply src/Features/CSharp/Portable/Organizing/Organizers/RecordDeclarationOrganizer.cs:40]: @CyrusNajmabadi Sounds reasonable. But I followed the same pattern used in other implementations in the same folder to be consistent with them. If I'd change it here, I'd change it everywhere in the same folder too. Do you want me to update all of them in this PR? or keep it for later PR?

Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/Organizing/Organizers/RecordDeclarationOrganizer.cs:40]: Two allocations is fine here :-)

Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/Organizing/Organizers/RecordDeclarationOrganizer.cs:40]: > I also would like to know where this feature exist in Visual Studio.

The feature never shipped. But it has come up recently as something we might like to resurrect

**Review by CyrusNajmabadi — APPROVED** [review]

**Pull request merged**